core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/resource/BaseTerminologyResourceCreateRequest.java

```diff
@@ -187,7 +208,52 @@ protected void preExecute(final TransactionContext context) {
 				.execute(context);
 		}
 	}
+	
+	@Override
+	@OverridingMethodsMustInvokeSuper
+	protected void checkParentCollection(TransactionContext context, Resource parentCollection) {
+		if (parentCollection instanceof TerminologyResourceCollection resourceCollection) {
+			// find the resource collection's tooling support for this child resource type
+			TerminologyResourceCollectionToolingSupport toolingSupport = context.service(TerminologyResourceCollectionToolingSupport.Registry.class).getToolingSupport(resourceCollection.getToolingId(), getResourceType());
+			
+			// append parent collection as a dependency of this new resource
+			final Dependency collectionDependency = Dependency.of(resourceCollection.getResourceURI());
+			setDependencies(dependencies == null ? List.of(collectionDependency) : ImmutableList.<Dependency>builder().addAll(dependencies).add(collectionDependency).build());
+			
+			// in case of matching child resource type, validate and inherit settings
+			inheritParentCollectionSettings(resourceCollection, toolingSupport.getInheritedSettingKeys());
+		} else if (parentCollection instanceof Bundle) {
+			// fetch all resources matching any of the current 
+			List<String> nonRootAncestorResourceIds = Collections3.toImmutableList(parentCollection.getResourcePathSegments()).stream().filter(path -> !IComponent.ROOT_ID.equals(path)).toList();
+			
+			if (!nonRootAncestorResourceIds.isEmpty()) {
+				// find and check the first matching terminology resource collection in the ancestor array (only one should exists)
+				ResourceRequests.prepareSearch()
+					.filterByResourceType(TerminologyResourceCollection.RESOURCE_TYPE)
+					.filterByIds(nonRootAncestorResourceIds)
+					.setLimit(nonRootAncestorResourceIds.size())
+					.build()
+					.execute(context)
+					.stream()
+					.filter(ancestorResource -> ancestorResource instanceof TerminologyResourceCollection)
+					.findFirst()
+					.ifPresent(ancestorTerminologyResourceCollection -> checkParentCollection(context, ancestorTerminologyResourceCollection));
+			}
+			
+		} else {
+			throw new BadRequestException("Selected parent resource '%s' is not a valid, recognizable collection resource.", parentCollection.getResourceURI());
+		}
+	}
 
+	private void inheritParentCollectionSettings(TerminologyResourceCollection resourceCollection, Set<String> inheritedSettingKeys) {
+		if (CompareUtils.isEmpty(resourceCollection.getSettings()) || CompareUtils.isEmpty(inheritedSettingKeys)) {
+			return;
+		}
+		
+		// take the filtered resource collection settings with the inherited keys and override it with the current settings if they are specified
+		setSettings(Json.assign(Maps.filterKeys(resourceCollection.getSettings(), inheritedSettingKeys::contains), getSettings()));
```

Comment:
Overridden key-value pairs should be logged, or also reported by throwing a `BadRequestException`.

Reply:
We allow overriding of ancestor collection settings values on per resource level (for the known inherited settings keys).